proof_of_concept/policy/replication.py

```diff
@@ -27,16 +32,17 @@ def __init__(self, namespace: str, key: RSAPublicKey) -> None:
     def is_valid(self, rule: Rule) -> bool:
         """Return True iff the rule is properly signed."""
         if isinstance(rule, ResultOfDataIn):
-            namespace = rule.data_asset[3:].split('.')[0]
+            namespace = rule.data_asset.split(':')[1]
         elif isinstance(rule, ResultOfComputeIn):
-            namespace = rule.compute_asset[3:].split('.')[0]
+            namespace = rule.compute_asset.split(':')[1]
         elif isinstance(rule, MayAccess):
-            namespace = rule.asset[3:].split('.')[0]
+            namespace = rule.asset.split(':')[1]
         elif isinstance(rule, InAssetCollection):
-            namespace = rule.asset[3:].split('.')[0]
+            namespace = rule.asset.split(':')[1]
         elif isinstance(rule, InPartyCollection):
-            namespace = rule.collection[3:].split('.')[0]
+            namespace = rule.collection.split(':')[1]
 
+        logger.info(f'Rule namespace {namespace}, should be {self._namespace}')
         if namespace != self._namespace:
             return False
```

Comment:
Can we somehow make use of the `AssetId.namespace` method here? Something like this (but then for all if/elif statements ofcourse)
```suggestion
            asset_id = AssetId(rule.collection)

        logger.info(f'Rule namespace {namespace}, should be {self._namespace}')
        if asset_id.namespace != self._namespace:
            return False
```

Reply:
Actually, I have a note about adding a `namespace()` method to Rule, which would clean this up even more nicely, but I decided not to pile that on here.

Reply:
Bueno